### ISSUE TITLE
`cryptographic-message-syntax`'s `SignedData::parse_ber` fails for some `ber`-encoded data

### ISSUE THREAD

**Issue by Hainish** [indygreg/PyOxidizer#521]
When calling `cryptographic-message-syntax`'s `SignedData::parse_ber` on the attached files, `CIARANG.RSA` parses correctly, but `1.RSA` does not, and results in an error:

```rust
DecodeErr(Malformed)
```

Parsing this file with `unber` works:

```shell
$ unber index-v1-guardian/META-INF/1.RSA | head
<C O="0" T="[UNIVERSAL 16]" TL="4" V="2295" A="SEQUENCE">
    <P O="4" T="[UNIVERSAL 6]" TL="2" V="9" A="OBJECT IDENTIFIER" F>1.2.840.113549.1.7.2</P>
    <C O="15" T="[0]" TL="4" V="2280">
        <C O="19" T="[UNIVERSAL 16]" TL="4" V="2276" A="SEQUENCE">
            <P O="23" T="[UNIVERSAL 2]" TL="2" V="1" A="INTEGER" F>1</P>
            <C O="26" T="[UNIVERSAL 17]" TL="2" V="11" A="SET">
                <C O="28" T="[UNIVERSAL 16]" TL="2" V="9" A="SEQUENCE">
                    <P O="30" T="[UNIVERSAL 6]" TL="2" V="5" A="OBJECT IDENTIFIER" F>1.3.14.3.2.26</P>
                    <P O="37" T="[UNIVERSAL 5]" TL="2" V="0" A="NULL"></P>
                </C O="39" T="[UNIVERSAL 16]" A="SEQUENCE" L="11">
```

[rsa-files.zip](https://github.com/indygreg/PyOxidizer/files/8218766/rsa-files.zip)

**Comment by Hainish** [comment]
POC: https://github.com/Hainish/parse-ber-poc

**Comment by indygreg** [comment]
Thanks for the report.

This looks like a bug in our parsing of the embedded X.509 certificate. Specifically with regards to handling of the `version` field of `TbsCertificate`.

Your provided reproduce file doesn't have the `version` field and our parser is choking on this. According to RFC 5280, this field is defined as `version [0]  EXPLICIT Version DEFAULT v1,` which means it is optional.

**Comment by indygreg** [comment]
I just published x509-certificate v0.8.0 and cryptographic-message-syntax v0.9.0 with this fix.

Thanks again for the bug report!
